### PR TITLE
Add support for a string phpcodesniffer-search-depth config value set via `composer config`

### PR DESCRIPTION
**Pull request by TravisCarden**
## Proposed Changes

Add support for a string value of the `extra.phpcodesniffer-search-depth` config value so it can be set from the command line (`composer config`).

## Related Issues

https://github.com/Dealerdirect/phpcodesniffer-composer-installer/issues/82

### CODE REVIEW

File: src/Plugin.php [hunk 1]
```diff
@@ -441,9 +441,9 @@ private function getMaxDepth()
             $maxDepth = $extra[self::KEY_MAX_DEPTH];
             $minDepth = $this->getMinDepth();
 
-            if (is_int($maxDepth) === false     /* Must be an integer */
-                || $maxDepth <= $minDepth       /* Larger than the minimum */
-                || is_float($maxDepth) === true /* Within the boundaries of integer */
+            if ((string) (int) $maxDepth !== (string) $maxDepth /* Must be an integer or cleanly castable to one */
```
Comment (Potherca) [inline src/Plugin.php]: I've never seen this before... Is there any benefit of this over using [`is_numeric`](https://www.php.net/manual/en/function.is-numeric.php)?

Comment (TravisCarden) [reply src/Plugin.php]: Yes, @Potherca. `is_numeric` would match floating point numbers, whereas we only want to match integers and string representations of them. There's no builtin function for that, as you know, but typecasting a string to an integer and back and then seeing whether it matches the original value is a common idiom for it. It's probably it's fairly intuitive once you've been introduced to it. :)

**Comment by Potherca** [comment]
To get Travis passing MR #86 needs to be merged.

**Comment by Potherca** [comment]
Awaiting MR #86, these changes can be seen to [pass the build in Travis](https://travis-ci.org/potherca-contrib/phpcodesniffer-composer-installer/builds/585267925).

**Comment by mjrider** [comment]
Could you please rebase this on master, because the build issues have been resolved

**Comment by TravisCarden** [comment]
Rebased, @mjrider.

**Pull request merged**